pkg/webhook/cspc.go

```diff
@@ -413,6 +413,23 @@ func validateBlockDevice(bd *openebsapis.BlockDevice, hostName string) error {
 			bd.Labels[types.HostNameLabelKey],
 		)
 	}
+
+	if bd.Labels[types.HostNameLabelKey] != hostName {
+		return errors.Errorf(
+			"block device %s doesn't belongs to node %s",
+			bd.Name,
+			bd.Labels[types.HostNameLabelKey],
+		)
+	}
```

Comment:
We already have a similar check above ^^

Reply:
Yes. I think rebasing brought this issue back. Fixed it. Thanks.